### PR TITLE
[AAP-7647] Support for in and contains in condition

### PR DESCRIPTION
**Pull request by mkanoor**
Added support for
* in
* not in
* contains
* not contains

https://issues.redhat.com/browse/AAP-7647

e.g.

   1. event.mylist contains "alert"
   2. event.name in ["alert","warning"]
   3. event.name not in ["error","warning"]
   4. event.mylist not contains "error"

This is a Drools only feature and wont work with durable rules

### CODE REVIEW

**Comment by Alex-Izquierdo** [comment]
closes https://issues.redhat.com/browse/AAP-5169

**Comment by Alex-Izquierdo** [comment]
Just out of curiosity, why two operator to perform the same operation? "contains" and "in" operators are not redundant? In ansible only exists the "in" way. 
From python zen: _There should be one-- and preferably only one --obvious way to do it._

**Comment by mkanoor** [comment]
@Alex-Izquierdo Its based on this https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_tests.html#testing-if-a-list-contains-a-value

"Ansible includes a **contains** test which operates similarly, but in reverse of the Jinja2 provided **in** test."

[1,2,3] contains 3
versus
3 in [1,2,3]

**Review by Alex-Izquierdo — APPROVED** [review]
I can not verify too much if the engine doesn't support it yet. I executed some regression tests without issues.

**Review by Alex-Izquierdo — APPROVED** [review]

**Comment by mkanoor** [comment]
@Alex-Izquierdo Examples added, the new drools_jpy jar has these operators

File: ansible_rulebook/json_generator.py [hunk 1]
```diff
@@ -161,6 +161,42 @@ def visit_condition(parsed_condition: ConditionTypes, variables: Dict):
                     ),
                 }
             }
+        elif parsed_condition.operator == "in":
+            return {
+                "ItemInListExpression": {
```
Comment (bzwei) [inline ansible_rulebook/json_generator.py:166]: Seems all the expressions having lhs and rhs have the same value. Can you extract them to share a function?

Comment (mkanoor) [reply ansible_rulebook/json_generator.py]: @bzwei made it into a function call.

**Review by benthomasson — APPROVED** [review]

File: ansible_rulebook/json_generator.py [hunk 2]
```diff
@@ -172,6 +188,15 @@ def visit_condition(parsed_condition: ConditionTypes, variables: Dict):
         raise Exception(f"Unhandled token {parsed_condition}")
 
 
+def create_binary_node(name, parsed_condition, variables):
+    return {
+        name: {
+            "lhs": visit_condition(parsed_condition.left, variables),
+            "rhs": (visit_condition(parsed_condition.right, variables)),
```
Comment (bzwei) [inline ansible_rulebook/json_generator.py:195]: extra () not needed

File: ansible_rulebook/json_generator.py [hunk 3]
```diff
@@ -161,6 +161,22 @@ def visit_condition(parsed_condition: ConditionTypes, variables: Dict):
                     ),
                 }
             }
+        elif parsed_condition.operator == "in":
+            return create_binary_node(
```
Comment (bzwei) [inline ansible_rulebook/json_generator.py:165]: can we refactor all existing applicable operators to use the same create_binary_node method?

File: ansible_rulebook/condition_parser.py [hunk 4]
```diff
@@ -158,4 +178,8 @@ def OperatorExpressionFactory(tokens):
 
 
 def parse_condition(condition_string: str) -> Condition:
-    return condition.parseString(condition_string)[0]
+    try:
+        return condition.parseString(condition_string, parse_all=True)[0]
+    except ParseException as pe:
+        print(pe.explain(depth=0))
```
Comment (bzwei) [inline ansible_rulebook/condition_parser.py:184]: why print to console not logger?

File: ansible_rulebook/json_generator.py (hunk 3 above)
Comment (mkanoor) [reply ansible_rulebook/json_generator.py:165]: @bzwei In a separate PR since its unrelated to these changes.

**Review by Alex-Izquierdo — APPROVED** [review]

**Review by bzwei — APPROVED** [review]

File: ansible_rulebook/json_generator.py (hunk 3 above)
Comment (mkanoor) [reply ansible_rulebook/json_generator.py:165]: @bzwei https://github.com/ansible/ansible-rulebook/pull/267

**Pull request merged**